backend/controller/lmevaljob_controller.go

```diff
@@ -186,40 +237,118 @@ func (r *LMEvalJobReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func (r *LMEvalJobReconciler) constructOptionsFromConfigMap(configmap *corev1.ConfigMap) error {
-	r.options.DriverImage = DefaultDriverImage
-	r.options.PodImage = DefaultPodImage
-	r.options.DriverServiceAccount = DefaultDriverServiceAccount
-	r.options.PodCheckingInterval = DefaultPodCheckingInterval
-	r.options.ImagePullPolicy = DefaultImagePullPolicy
-	log := log.FromContext(context.Background())
+func (r *LMEvalJobReconciler) checkSecrets(ctx context.Context) error {
 
-	if v, found := configmap.Data[DriverImageKey]; found {
-		r.options.DriverImage = v
+	var isSecretExists = func(name string) bool {
+		var secret corev1.Secret
+		err := r.Get(ctx, types.NamespacedName{Namespace: r.Namespace, Name: name}, &secret)
+		return err == nil
 	}
-	if v, found := configmap.Data[PodImageKey]; found {
-		r.options.PodImage = v
+
+	if !isSecretExists(r.options.GrpcServerSecret) {
+		return fmt.Errorf("secret %s not found", r.options.GrpcServerSecret)
 	}
-	if v, found := configmap.Data[DriverServiceAccountKey]; found {
-		r.options.DriverServiceAccount = v
+	if !isSecretExists(r.options.GrpcClientSecret) {
+		return fmt.Errorf("secret %s not found", r.options.GrpcServerSecret)
 	}
-	if v, found := configmap.Data[PodCheckingIntervalKey]; found {
-		if d, err := time.ParseDuration(v); err == nil {
-			r.options.PodCheckingInterval = d
-		} else {
-			log.Error(err, "failed to parse the configmap", PodCheckingIntervalKey, v)
-		}
+	return nil
+}
+
+func (r *LMEvalJobReconciler) updateStatus(ctx context.Context, newStatus *backendv1beta1.JobStatus) error {
+	log := log.FromContext(ctx)
+
+	if newStatus.GetJobName() == "" || newStatus.GetJobNamespace() == "" {
+		return fmt.Errorf("JobName or JobNameSpace is empty")
 	}
-	if v, found := configmap.Data[ImagePullPolicyKey]; found {
-		if p, found := pullPolicyMap[corev1.PullPolicy(v)]; found {
-			r.options.ImagePullPolicy = p
-		} else {
-			log.Error(
-				fmt.Errorf("invalid %s value in the configmap: %s", ImagePullPolicyKey, v),
-				"use the default value for the ImagePullPolicy instead",
-			)
+
+	job := &lmevalservicev1beta1.LMEvalJob{}
+	if err := r.Get(ctx, types.NamespacedName{
+		Namespace: newStatus.JobNamespace,
```

Comment:
when constructing the controller, it could monitor the CR under a specific namespace or all namespaces. it depends on how users plan to use it. But this reminds me that we should have two sets of manifests, one for the clusterrole and one for the role. ah... more work to do

Reply:
created an issue for this #22